From 4455f8da880386b2d4c66fb423b7d87d12462c18 Mon Sep 17 00:00:00 2001
From: Tarulia <mihawk.90+git@googlemail.com>
Date: Mon, 6 Oct 2025 20:59:33 +0200
Subject: [PATCH] plugins: Set missed default bitrates to 6000 kbps

efca325 already increased the default bitrate for various encoders.
However, some were missed and this commit fills the gap:

* Mac-VT
* OpenH264
* Texture AMF
* VAAPI
---
 plugins/mac-videotoolbox/encoder.c       | 4 ++--
 plugins/obs-ffmpeg/obs-ffmpeg-openh264.c | 2 +-
 plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c    | 2 +-
 plugins/obs-ffmpeg/texture-amf.cpp       | 8 ++++----
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/plugins/mac-videotoolbox/encoder.c b/plugins/mac-videotoolbox/encoder.c
index 5d9d9e4ca4ca60..acd94c6983df83 100644
--- a/plugins/mac-videotoolbox/encoder.c
+++ b/plugins/mac-videotoolbox/encoder.c
@@ -1375,10 +1375,10 @@ static void vt_defaults(obs_data_t *settings, void *data)
 			obs_data_set_default_string(settings, "rate_control", "CBR");
 		}
 	}
-	obs_data_set_default_int(settings, "bitrate", 2500);
+	obs_data_set_default_int(settings, "bitrate", 6000);
 	obs_data_set_default_int(settings, "quality", 60);
 	obs_data_set_default_bool(settings, "limit_bitrate", false);
-	obs_data_set_default_int(settings, "max_bitrate", 2500);
+	obs_data_set_default_int(settings, "max_bitrate", 6000);
 	obs_data_set_default_double(settings, "max_bitrate_window", 1.5f);
 	obs_data_set_default_int(settings, "keyint_sec", 2);
 	obs_data_set_default_string(settings, "profile",
diff --git a/plugins/obs-ffmpeg/obs-ffmpeg-openh264.c b/plugins/obs-ffmpeg/obs-ffmpeg-openh264.c
index 6a56fe57118174..f2b28c2e99036f 100644
--- a/plugins/obs-ffmpeg/obs-ffmpeg-openh264.c
+++ b/plugins/obs-ffmpeg/obs-ffmpeg-openh264.c
@@ -193,7 +193,7 @@ static bool openh264_encode(void *data, struct encoder_frame *frame, struct enco
 
 void openh264_defaults(obs_data_t *settings)
 {
-	obs_data_set_default_int(settings, "bitrate", 2500);
+	obs_data_set_default_int(settings, "bitrate", 6000);
 	obs_data_set_default_string(settings, "profile", "main");
 }
 
diff --git a/plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c b/plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c
index 38c9506d62e3d2..32d5f4fa5dbe49 100644
--- a/plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c
+++ b/plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c
@@ -861,7 +861,7 @@ static void vaapi_defaults_internal(obs_data_t *settings, enum codec_type codec)
 	else if (codec == CODEC_AV1)
 		obs_data_set_default_int(settings, "profile", AV_PROFILE_AV1_MAIN);
 	obs_data_set_default_int(settings, "level", AV_LEVEL_UNKNOWN);
-	obs_data_set_default_int(settings, "bitrate", 2500);
+	obs_data_set_default_int(settings, "bitrate", 6000);
 	obs_data_set_default_int(settings, "keyint_sec", 0);
 	obs_data_set_default_int(settings, "bf", 0);
 	obs_data_set_default_int(settings, "qp", 20);
diff --git a/plugins/obs-ffmpeg/texture-amf.cpp b/plugins/obs-ffmpeg/texture-amf.cpp
index 7013df62d9fde0..af9a5a3c5b6365 100644
--- a/plugins/obs-ffmpeg/texture-amf.cpp
+++ b/plugins/obs-ffmpeg/texture-amf.cpp
@@ -1383,7 +1383,7 @@ static inline void adjust_recommended_avc_defaults(amf_base *enc, obs_data_t *se
 	int64_t framerate = enc->fps_num / enc->fps_den;
 	if ((enc->cx * enc->cy > 1920 * 1088) || (framerate > 60)) {
 		// Recommended base defaults
-		obs_data_set_default_int(settings, "bitrate", 2500);
+		obs_data_set_default_int(settings, "bitrate", 6000);
 		obs_data_set_default_int(settings, "cqp", 20);
 		obs_data_set_default_string(settings, "rate_control", "CBR");
 		obs_data_set_default_string(settings, "preset", "quality");
@@ -1841,7 +1841,7 @@ static inline void adjust_recommended_hevc_defaults(amf_base *enc, obs_data_t *s
 	const int64_t framerate = enc->fps_num / enc->fps_den;
 	if ((enc->cx * enc->cy > 1920 * 1088) || is10bit || (framerate > 60)) {
 		// Recommended base defaults
-		obs_data_set_default_int(settings, "bitrate", 2500);
+		obs_data_set_default_int(settings, "bitrate", 6000);
 		obs_data_set_default_int(settings, "cqp", 20);
 		obs_data_set_default_string(settings, "preset", "quality");
 		info("Original base default settings were used according to resolution and framerate.");
@@ -2092,7 +2092,7 @@ try {
 static void amf_hevc_defaults(obs_data_t *settings)
 {
 	obs_data_set_default_string(settings, "rate_control", "CBR");
-	obs_data_set_default_int(settings, "bitrate", 2500);
+	obs_data_set_default_int(settings, "bitrate", 6000);
 	obs_data_set_default_int(settings, "cqp", 20);
 	obs_data_set_default_string(settings, "preset", "quality");
 }
@@ -2245,7 +2245,7 @@ static inline void adjust_recommended_av1_defaults(amf_base *enc, obs_data_t *se
 	const int64_t framerate = enc->fps_num / enc->fps_den;
 	if ((enc->cx * enc->cy > 1920 * 1088) || is10bit || (framerate > 60)) {
 		// Recommended base defaults
-		obs_data_set_default_int(settings, "bitrate", 2500);
+		obs_data_set_default_int(settings, "bitrate", 6000);
 		obs_data_set_default_int(settings, "cqp", 20);
 		obs_data_set_default_string(settings, "preset", "balanced");
 		obs_data_set_default_string(settings, "profile", "main");
